component/addons/openshift-certs.libsonnet

```
@@ -0,0 +1,79 @@
+local kube = import 'lib/kube.libjsonnet';
+local rl = import 'lib/resource-locker.libjsonnet';
+
+local sourceSecretNamespace = 'openshift-monitoring';
+local sourceSecretName = 'metrics-client-certs';
+
+{
+  local config = self,
+
+  local targetSecret = kube.Secret('ocp-metric-client-certs-' + config.values.prometheus.name),
+
+  local rlPatch = [
+    if o.kind == 'ResourceLocker' then
+      o {
+        spec+: {
+          resources: [
+            {
+              object: {
+                apiVersion: targetSecret.apiVersion,
+                kind: targetSecret.kind,
+                metadata: {
+                  name: targetSecret.metadata.name,
+                  namespace: config.values.common.namespace,
+                },
+                data: {},
+              },
+              excludedPaths: [
+                // We patch .data in the patch definition
+                '.data',
+                // Need to be here or we get an argocd diff. The resoucre locker operator will add them.
+                '.spec.replicas',
+                '.metadata',
+                '.status',
+              ],
+            },
+          ],
+          patches: [
+            {
+              id: 'patch-1',
+              targetObjectRef: {
+                apiVersion: targetSecret.apiVersion,
+                kind: targetSecret.kind,
+                name: targetSecret.metadata.name,
+                namespace: config.values.common.namespace,
+              },
+              patchTemplate: |||
+                data:
+                  tls.crt: {{ index (index . 0).data "tls.crt" }}
+                  tls.key: {{ index (index . 0).data "tls.key" }}
+              |||,
+              patchType: 'application/strategic-merge-patch+json',
+              sourceObjectRefs: [
+                {
+                  apiVersion: targetSecret.apiVersion,
+                  kind: targetSecret.kind,
+                  name: sourceSecretName,
+                  namespace: sourceSecretNamespace,
+                },
+              ],
+            },
+          ],
```

Instead of overwriting the preconfigured patch you could just edit it since you know you'll get an array with exactly one element from `rl.Patch()`:

```suggestion
          patches: [
            super.patches[0] {
              sourceObjectRefs: [
                {
                  apiVersion: targetSecret.apiVersion,
                  kind: targetSecret.kind,
                  name: sourceSecretName,
                  namespace: sourceSecretNamespace,
                },
              ],
            },
          ],
```